### ISSUE TITLE
Inconsistent behaviour with custom model methods

### ISSUE THREAD

**Issue by robertklep** [pstaender/neo4jmapper#20]
I'm using the following script as a simplified test case for some of my code:

```
var Neo4j = require('neo4jmapper');
var neo4j = new Neo4j('http://127.0.0.1:7474');
var Node  = neo4j.Node;

var ModelA = Node.registerModel('ModelA');
ModelA.prototype.test = function() { console.log('Hello, world!'); };

var ModelB = ModelA.registerModel('ModelB');

var modela = new ModelA({ title : 'foo' });
var modelb = new ModelB({ title : 'bar' });

console.log('Model A            :', modela.test !== undefined);
console.log('Model B before save:', modelb.test !== undefined);

modelb.save(function(err, doc) {
  if (err) throw err;
  console.log('Model B after save :', doc.test !== undefined);
  ModelB.findOne({ title : 'bar' }, function(err, doc2) {
    if (err) throw err;
    console.log('Model B findOne    :', doc2.test !== undefined);
  });
  ModelB.findById(doc.id, function(err, doc3) {
    if (err) throw err;
    console.log('Model B findById   :', doc3.test !== undefined);
  });
});
```

**TL;DR:** it shows if the model instance has a `test()` method after certain performed actions.

The output is inconsistent between different versions of neo4jmapper:

```
#1.0.0-beta11
Model A            : true
Model B before save: true
Model B after save : true
Model B findById   : false # ???
Model B findOne    : true

#1.0.0-beta12 (and 1.0.0-rc1 as well)
Model A            : true
Model B before save: true
Model B after save : true
Model B findById   : false
Model B findOne    : false
```

Also, it's especially strange that it works for `findOne` for beta11, but that it fails for `findById`.
